examples/failure-cases/forbid-dependency-resolution-fork/smithy-build.json

```diff
@@ -0,0 +1,11 @@
+{
+  "version": "1.0",
+  "maven": {
+    "dependencies": [
+      "software.amazon.smithy:smithy-aws-traits:[1.0,2.0)"
```

Comment:
Per: https://docs.gradle.org/current/userguide/single_versions.html
This will exclude all versions starting with 2.0 that are smaller than 2.0 (so it allows 2.0.1 but not 2.0-dev). We dont really use snapshot releases so it isnt super relevant, but its an ok convention to follow.